### ISSUE TITLE
🐛 Code splitting should duplicate modules used by web workers

### ISSUE THREAD

**Issue by bwrrp** [parcel-bundler/parcel#758]
**Choose one:** this is a 🐛 bug report

Web workers run in an entirely separate JS context. It seems that the current approach treats a `new Worker(path)` expression as a normal code splitting point. The result of this is that modules shared by multiple workers end up in the main bundle instead of in each worker's `.js` file.

### 🎛 Configuration (.babelrc, package.json, cli command)

Seen on a very minimal project, see example code below. Create those files, install parcel-bundler and run `npx parcel index.html --no-hmr`. Disabling HMR is not necessary to trigger this behavior, but it makes the bundled code a bit easier to read.

### 🤔 Expected Behavior

I expected the code from `shared.js` to end up duplicated in each web worker's .js bundle.

### 😯 Current Behavior

The code for `shared.js` is included in the `parcel.js` bundle running in the web page context. The bundles generated for `w1.js` and `w2.js` contain the require call for './shared', but do not contain the module itself. This results in the following error being thrown in each of the workers:

```
Uncaught Error: Cannot find module '12'
newRequire	@	d2f19dd….js:33
localRequire	@	d2f19dd….js:48
require.9../shared	@	w2.js:1
newRequire	@	d2f19dd….js:42
require.9../shared	@	d2f19dd….js:67
(anonymous)	@	d2f19dd….js:72
```

### 💁 Possible Solution

Workers should probably considered a special case in code splitting, as all modules required by one should actually be duplicated into the resulting bundle.

### 🔦 Context

I'm building an application that requires communication between several web workers. These workers have different responsibilities, and therefore a different entry point, but do share large parts of the codebase. I don't really want to maintain a separate copy of these shared modules for each worker.

### 💻 Code Sample

*index.html*
```html
<script src="index.js"></script>
```

*index.js*
```js
new Worker("./w1.js");
new Worker("./w2.js");
```

*w1.js*
```js
import shared from "./shared";
console.log("w1", shared);
```

*w2.js*
```js
import shared from "./shared";
console.log("w2", shared);
```

*shared.js*
```js
export default { test: 123 }
```

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 5.6.0
| Node             | 8.1.3
| npm         | 5.6.0
| Operating System | Linux

**Comment by jsilveira** [comment]
Is there a workaround for this issue? I'm having the same problem and I can't figure out a way to patch it until a permanent solution is found. Thanks!

**Comment by termhn** [comment]
This seems to be related and it's very puzzling to me. See: https://github.com/termhn/nanofaucet-client

On `master` branch, the webworkers are bundled properly and work. On `unistore-router` branch, with no (seemingly) functional changes to how the workers themselves are used, this "cannot find module" error occurs.

While you can't actually use the resulting app here without access to the backend for it, you should be able to `yarn install` and `yarn start` on each branch and see that the generated javascript file for the webworker on each are vastly different in size: one around 5k lines and one around 28k. This seems to be because of the bug above, but, as @jsilveira said, I'm not sure why this occurs or how to patch it yet. Just thought an example of it working and of it not working might be helpful in figuring that out.

**Comment by termhn** [comment]
cc @jdanyow any ideas here?

**Comment by jdanyow** [comment]
@termhn this is another scenario that requires #189

**Comment by termhn** [comment]
@jdanyow Hmm, really? Seems puzzling since it seemingly can work properly now just unreliably... Why would it be working with https://github.com/termhn/nanofaucet-client on one branch but not the other?

**Comment by mischnic** [comment]
Ideally, code wouldn't have to be downloaded twice (or more times). This could be achieved by splitting the shared code into one bundle, which get's imported twice but can be cached.

See: https://github.com/webpack/webpack/issues/6472

**Comment by BrieucP** [comment]
Using importScript looks for me like the closest solution to the js standard. 
Import should not duplicate code.
Import script should not be bundled but should be treated like worker and service worker.

**Comment by devongovett** [comment]
Fixed by #1257.

Ideally the shared modules could be placed into a separate bundle as has been stated in this thread. In order to be consistent about how `require()` works, we won't make it automatically create a split point when called in a worker that would cause the code to potentially be removed from another bundle. But we could make this work if you use an async `import()` in workers for code splitting. That isn't supported yet, but should be easy enough to add.

**Comment by devongovett** [comment]
Made a new issue for the second part: #1654
